sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/MetricReaderFactory.java

```diff
@@ -77,39 +71,15 @@ public MetricReader create(
       }
       Prometheus prometheusModel = exporterModel.getPrometheus();
       if (prometheusModel != null) {
-        // Translate from file configuration scheme to environment variable scheme. This is
-        // ultimately
-        // interpreted by PrometheusMetricReaderProvider, but we want to avoid the dependency on
-        // opentelemetry-exporter-prometheus
-        Map<String, String> properties = new HashMap<>();
-        if (prometheusModel.getHost() != null) {
-          properties.put("otel.exporter.prometheus.host", prometheusModel.getHost());
-        }
-        if (prometheusModel.getPort() != null) {
-          properties.put(
-              "otel.exporter.prometheus.port", String.valueOf(prometheusModel.getPort()));
-        }
-
-        ConfigProperties configProperties = DefaultConfigProperties.createFromMap(properties);
-        return FileConfigUtil.addAndReturn(
-            closeables,
-            FileConfigUtil.assertNotNull(
-                metricReaderSpiManager(configProperties, spiHelper).getByName("prometheus"),
-                "prometheus reader"));
+        MetricReader metricReader =
+            FileConfigUtil.loadComponent(
+                spiHelper, MetricReader.class, "prometheus", prometheusModel);
+        return FileConfigUtil.addAndReturn(closeables, metricReader);
```

Comment:
Here's what it looks like to load a component in the file configuration implementation. We state that we want ` ComponentProvider` which provides instances of `MetricReader.class` named `prometheus`, and we pass the `prometheusModel` to it (which is translated to StructuredConfigProperties).

Nice and easy.